twocaptchaapi/__init__.py

```diff
@@ -127,33 +129,75 @@ def get_load(self):
         """Obtains load statistics of the server."""
         return self.get(self.LOAD_URL, {}).text
 
+    def is_base64(self, string):
+        try:
+            base64.decodestring(string)
+            return True
+        except (binascii.Error):
+            return False
+
     @_rewrite_http_to_com_err
     @_rewrite_to_format_err(IndexError, ValueError)
     def solve(self, file, captcha_parameters=None):
         """
-        Queues a captcha for solving. `file` may either be a path or a file object.
+        Queues a captcha for solving. `file` may either be a path, a file object or a base64 string.
         Optional parameters for captcha solving may be specified in a `dict` via
         `captcha_parameters`, for valid values see section "Additional CAPTCHA parameters"
         in API documentation here:
-
         https://2captcha.com/api-2captcha
         """
-
-        # If path was provided, load file.
-        if type(file) == str:
-            with open(file, 'rb') as f:
-                raw_data = f.read()
+        if self.is_base64(file):
+            # Send request.
+            text = self.post(
+                self.REQ_URL,
+                data = {'key': self.api_key, 'method': 'base64', 'body': file}
+            ).text
         else:
-            raw_data = file.read()
+            # If path was provided, load file.
+            if type(file) == str:
+                with open(file, 'rb') as f:
+                    raw_data = f.read()
+            else:
+                raw_data = file.read()
+
+            # Detect image format.
+            file_ext = imghdr.what(None, h=raw_data)
+
+            # Send request.
+            text = self.post(
+                self.REQ_URL,
+                captcha_parameters or {'method': 'post'},
+                files={'file': ('captcha.' + file_ext, raw_data)}
+            ).text
+
+        # Success?
```

Comment:
Possibly put that function tail into an extra internal module scope function `_captcha_from_response` to avoid duplication of the logic? What do you think?